README


This archive contains the distribution Role-REST-Client-Auth-Basic,
version 0.04:

  Basic Authentication for REST Client Role

This software is copyright (c) 2012 by Kaare Rasmussen.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Role-Model-REST-Auth-Basic

0.04    2012-03-30 11:10:27 Europe/Copenhagen
 - Don't use 5.010. We don't need it.

0.03    2012-03-22 07:09:32 Europe/Copenhagen
 - httpheaders require Role::REST::Client 0.03

0.02    2012-03-21 08:08:59 Europe/Copenhagen
 - _headers changes to httpheaders in Role::REST::Client

0.01    2012-03-01 21:20:18 Europe/Copenhagen
 - First version
   Thanks goes to Alex Howarth for testing and patching


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Module::Build" => "0.3601",
    "Moose" => 0,
    "Role::REST::Client" => "0.03",
    "Test::More" => "0.88",
    "strict" => 0,
    "warnings" => 0
  },
  "configure_requires" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Basic Authentication for REST Client Role",
  "dist_author" => [
    "Kaare Rasmussen <kaare at cpan dot net>"
  ],
  "dist_name" => "Role-REST-Client-Auth-Basic",
  "dist_version" => "0.04",
  "license" => "perl",
  "module_name" => "Role::REST::Client::Auth::Basic",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "MIME::Base64" => 0,
    "Moose::Role" => 0
  },
  "script_files" => []
);


my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Role/REST/Client/Auth/Basic.pm
package Role::REST::Client::Auth::Basic;
{
  $Role::REST::Client::Auth::Basic::VERSION = '0.04';
}

use Moose::Role;
use MIME::Base64;

requires '_call', 'httpheaders';

has 'user' => (
	isa => 'Str',
	is  => 'rw',
    predicate => 'has_user',
	trigger => sub {
		my ($self, $user) = @_;
		Carp::croak("Basic authentication user name can't contain ':'") if $user =~ /:/;
	},
);
has 'passwd' => (
	isa => 'Str',
	is  => 'rw',
    predicate => 'has_passwd',
);

before '_call' => sub {
	my ($self, $method, $endpoint, $data, $args) = @_;
	return if $args->{authentication} and $args->{authentication} ne 'basic';

	if ($self->has_user) {
		my $user = $self->user;
		my $passwd = $self->has_passwd ? $self->passwd : '';
		$self->set_header('Authorization', 'Basic ' . MIME::Base64::encode("$user:$passwd", ''));
	}
	return;
};

1;



=pod

=head1 NAME

Role::REST::Client::Auth::Basic - Basic Authentication for REST Client Role

=head1 VERSION

version 0.04

=head1 SYNOPSIS

	{
		package RESTExample;

		use Moose;
		with 'Role::REST::Client';
		with 'Role::REST::Client::Auth::Basic';

		sub bar {
			my ($self) = @_;
			my $res = $self->post('foo/bar/baz', {foo => 'bar'});
			my $code = $res->code;
			my $data = $res->data;
			return $data if $code == 200;
	   }

	}

	my $foo = RESTExample->new( 
		server =>      'http://localhost:3000',
		type   =>      'application/json',
		user   =>      'mee',
		passwd =>      'sekrit',
	);

	$foo->bar;

	# controller
	sub foo : Local {
		my ($self, $c) = @_;
		# Call w/ basic authentication
		my $res = $c->model('MyData')->post('foo/bar/baz', {foo => 'bar'});
		my $code = $res->code;
		my $data = $res->data;
		...
		# Call w/o basic authentication
		my $res = $c->model('MyData')->post('xyzzy', {foo => 'bar'}, {authentication => undef});
	}

=head1 DESCRIPTION

This role adds basic authentication to Role::REST::Client.

Just add it to your class and all calls will automatically authenticate.

Add an authentication parameter to the arguments if you for some reaon don't want to authenticate

=head1 NAME

Role::REST::Client::Auth::Basic - Basic Authentication for REST Client Role

=head1 AUTHOR

Kaare Rasmussen, <kaare at cpan dot com>

=head1 BUGS 

Please report any bugs or feature requests to bug-role-rest-client-auth-basic at rt.cpan.org, or through the
web interface at http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Role-REST-Client-Auth-Basic.

=head1 COPYRIGHT & LICENSE 

Copyright 2012 Kaare Rasmussen, all rights reserved.

This library is free software; you can redistribute it and/or modify it under the same terms as 
Perl itself, either Perl version 5.8.8 or, at your option, any later version of Perl 5 you may 
have available.

=head1 AUTHOR

Kaare Rasmussen <kaare at cpan dot net>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2012 by Kaare Rasmussen.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


__END__

# ABSTRACT: Basic Authentication for REST Client Role



Makefile.PL

use strict;
use warnings;



use ExtUtils::MakeMaker 6.30;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Basic Authentication for REST Client Role",
  "AUTHOR" => "Kaare Rasmussen <kaare at cpan dot net>",
  "BUILD_REQUIRES" => {
    "File::Find" => 0,
    "File::Temp" => 0,
    "Module::Build" => "0.3601",
    "Moose" => 0,
    "Role::REST::Client" => "0.03",
    "Test::More" => "0.88",
    "strict" => 0,
    "warnings" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.30",
    "Module::Build" => "0.3601"
  },
  "DISTNAME" => "Role-REST-Client-Auth-Basic",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Role::REST::Client::Auth::Basic",
  "PREREQ_PM" => {
    "MIME::Base64" => 0,
    "Moose::Role" => 0
  },
  "VERSION" => "0.04",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



